python/langsmith/client.py

```diff
@@ -4664,6 +4664,538 @@ def _evaluate_strings(
             **kwargs,
         )
 
+    def _get_settings(self) -> dict:
+        """Get the settings for the current tenant.
+
+        Returns:
+            dict: The settings for the current tenant.
+        """
+        response = self.request_with_retries("GET", "/settings")
+        return response.json()
+
+    def _current_tenant_is_owner(self, owner: str) -> bool:
+        """Check if the current workspace has the same handle as owner.
+
+        Args:
+            owner (str): The owner to check against.
+
+        Returns:
+            bool: True if the current tenant is the owner, False otherwise.
+        """
+        settings = self._get_settings()
+        return owner == "-" or settings["tenant_handle"] == owner
+
+    def _owner_conflict_error(
+        self, action: str, owner: str
+    ) -> ls_utils.LangSmithUserError:
+        return ls_utils.LangSmithUserError(
+            f"Cannot {action} for another tenant.\n"
+            f"Current tenant: {self._get_settings()['tenant_handle']},\n"
+            f"Requested tenant: {owner}"
+        )
+
+    def _get_latest_commit_hash(
+        self, prompt_owner_and_name: str, limit: int = 1, offset: int = 0
+    ) -> Optional[str]:
+        """Get the latest commit hash for a prompt.
+
+        Args:
+            prompt_owner_and_name (str): The owner and name of the prompt.
+            limit (int): The maximum number of commits to fetch. Defaults to 1.
+            offset (int): The number of commits to skip. Defaults to 0.
+
+        Returns:
+            Optional[str]: The latest commit hash, or None if no commits are found.
+        """
+        response = self.request_with_retries(
+            "GET",
+            f"/commits/{prompt_owner_and_name}/",
+            params={"limit": limit, "offset": offset},
```

Comment:
ooc: so we're guaranteed to always get these in reverse order by time? Would that ever change?

Reply:
right now it's hardcoded in the backend as `ORDER BY created_at DESC`
should probably accept sort type and direction params, but will do that at a later date